modules/phase_field/src/kernels/MatReaction.C

```diff
@@ -0,0 +1,77 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+#include "MatReaction.h"
+
+template<>
+InputParameters validParams<MatReaction>()
+{
+  InputParameters params = validParams<Kernel>();
+  params.addCoupledVar("v", "Set this to make v a coupled variable, otherwise it will use the kernel's nonlinear variable for v");
+  params.addClassDescription("Kernel to add -L*v, where L=reaction rate, v= variable");
+  params.addParam<MaterialPropertyName>("mob_name", "L", "The reaction rate used with the kernel");
+  params.addCoupledVar("args", "Vector of nonlinear variable arguments this object depends on");
+  return params;
+}
+
+MatReaction::MatReaction(const InputParameters & parameters) :
+  DerivativeMaterialInterface<JvarMapInterface<Kernel> >(parameters),
```

Comment:
this line should also be indented by 4 spaces (and the class description has an inconsistent whitespace after the = sign ... being really picky here)